bigquery/google/cloud/bigquery/job.py

```diff
@@ -1277,6 +1278,22 @@ def from_api_repr(cls, resource, client):
         job._set_properties(resource)
         return job
 
+    @property
+    def query_plan(self):
+        """Return query plan from job statistics, if present.
+
+        See:
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#statistics.query.queryPlan
+
+        :rtype: list of dict
```

Comment:
Feels a bit odd to me to return a dict here rather than a new object, but seems okay for the first pass.